src/silo/config/runtime_config.cpp

```diff
@@ -4,50 +4,183 @@
+               "lazily (by streaming)."
+            ),
+         }
+   };
+}
+
+RuntimeConfig::RuntimeConfig() {
+   overwriteFrom(getConfigSpecification().getConfigSourceFromDefaults());
+}
+
+std::vector<std::filesystem::path> RuntimeConfig::getConfigPaths() const {
+   std::vector<std::filesystem::path> result;
+   if (default_runtime_config.has_value()) {
+      result.emplace_back(default_runtime_config.value());
+   }
+   if (runtime_config.has_value()) {
+      result.emplace_back(runtime_config.value());
+   }
+   return result;
+}
+
+void RuntimeConfig::overwriteFrom(const VerifiedConfigAttributes& config_source) {
+   if (auto var = config_source.getPath(runtimeConfigOptionKey())) {
+      runtime_config = var.value();
+   }
+   if (auto var = config_source.getPath(defaultRuntimeConfigOptionKey())) {
+      default_runtime_config = var.value();
+   }
+   if (auto var = config_source.getPath(dataDirectoryOptionKey())) {
+      data_directory = var.value();
+   }
+   if (auto var = config_source.getUint16(apiPortOptionKey())) {
+      api_options.port = var.value();
+   }
+   if (auto var = config_source.getInt32(apiMaxConnectionsOptionKey())) {
+      api_options.max_connections = var.value();
+   }
+   if (auto var = config_source.getInt32(apiParallelThreadsOptionKey())) {
+      api_options.parallel_threads = var.value();
+   }
+   // But estimated_startup_end is a derived value:
+   if (auto var = config_source.getUint32(apiEstimatedStartupTimeOptionKey())) {
+      const std::chrono::minutes minutes = std::chrono::minutes(var.value());
       api_options.estimated_startup_end = std::chrono::system_clock::now() + minutes;
    }
+   if (auto var = config_source.getUint32(queryMaterializationOptionKey())) {
+      query_options.materialization_cutoff = var.value();
+   }
 }
 
 }  // namespace silo::config
+
+namespace silo::common {
+// Using a macro to get the declared variable name #FIELD_NAME
+#define ADD_FIELD_TO_RESULT(VARIABLE, FIELD_NAME) \
+   result += fmt::format("{}: {}, ", #FIELD_NAME, toDebugString(VARIABLE.FIELD_NAME))
+
+std::string toDebugString(const silo::config::ApiOptions& api_options) {
+   std::string result = "{";
+   ADD_FIELD_TO_RESULT(api_options, max_connections);
+   ADD_FIELD_TO_RESULT(api_options, parallel_threads);
+   ADD_FIELD_TO_RESULT(api_options, port);
+   ADD_FIELD_TO_RESULT(api_options, estimated_startup_end);
+   result += "}";
+   return result;
+}
+
+std::string toDebugString(const silo::config::QueryOptions& query_options) {
+   std::string result = "{";
+   ADD_FIELD_TO_RESULT(query_options, materialization_cutoff);
+   result += "}";
+   return result;
+}
+
+std::string toDebugString(const silo::config::RuntimeConfig& runtime_config) {
+   std::string result = "{";
+   ADD_FIELD_TO_RESULT(runtime_config, data_directory);
+   ADD_FIELD_TO_RESULT(runtime_config, api_options);
+   ADD_FIELD_TO_RESULT(runtime_config, query_options);
+   result += "}";
+   return result;
+}
```

Comment:
My claim is that by using this, we could simplify this PR and add less code. We can also do it in this PR.

Reply:
If alone, I would refuse because it would take time for me to understand and change and that would delay the PR and I was asked to submit PRs as soon as possible (that they have some more lines of code makes them worse, but the point has always been to make them quicker). But let me discuss it with @Taepper

Reply:
I tried it and it actually worked pretty good. We put it into the nlohmann namespace also resolving more review comments